datahub-web-react/src/app/entity/user/UserInfoSideBar.tsx

```diff
@@ -0,0 +1,347 @@
+ */
+export default function UserInfoSideBar() {
+    const { urn: encodedUrn } = useUserParams();
+    const urn = decodeUrn(encodedUrn);
+    const [updateCorpUserPropertiesMutation] = useUpdateCorpUserPropertiesMutation();
+    const { loading, error, data, refetch } = useGetUserQuery({ variables: { urn, groupsCount: GROUP_PAGE_SIZE } });
+    const entityRegistry = useEntityRegistry();
+
+    const [groupSectionExpanded, setGroupSectionExpanded] = useState(false);
+    const [editProfileModal, showEditProfileModal] = useState(false);
+    /* eslint-disable @typescript-eslint/no-unused-vars */
+    const [editableAboutMeText, setEditableAboutMeText] = useState<string | undefined>('');
+
+    const groupsDetails = data?.corpUser?.relationships as ExtendedEntityRelationshipsResult;
+    const photoUrl = data?.corpUser?.editableProperties?.pictureLink || undefined;
+
+    if (error || (!loading && !error && !data)) {
+        return <Alert type="error" message={error?.message || 'Profile data failed to load'} />;
+    }
+
+    // EditProfile modal Constants
+    const getEditModalData = () => {
+        return {
+            urn: data?.corpUser?.urn || undefined,
+            name: data?.corpUser?.editableProperties?.displayName || data?.corpUser?.info?.fullName || undefined,
+            title: data?.corpUser?.editableProperties?.title || data?.corpUser?.info?.title || undefined,
+            team: data?.corpUser?.editableProperties?.teams?.join(',') || undefined,
+            email: data?.corpUser?.editableProperties?.email || data?.corpUser?.info?.email || undefined,
+            image: photoUrl,
+            slack: data?.corpUser?.editableProperties?.slack || undefined,
+            phone: data?.corpUser?.editableProperties?.phone || undefined,
+        };
+    };
+
+    // About Text save
+    const onSaveAboutMe = (inputString) => {
+        setEditableAboutMeText(inputString);
+        updateCorpUserPropertiesMutation({
+            variables: {
+                urn: data?.corpUser?.urn || '',
+                input: {
+                    aboutMe: inputString,
+                },
+            },
+        })
+            .catch((e) => {
+                message.destroy();
+                message.error({ content: `Failed to Save changes!: \n ${e.message || ''}`, duration: 3 });
+            })
+            .finally(() => {
+                message.success({
+                    content: `Changes saved.`,
+                    duration: 3,
+                });
+                refetch();
+            });
+    };
+    return (
+        <>
+            {loading && <Message type="loading" content="Loading..." style={MESSAGE_STYLE} />}
+            <SideBar>
+                <SideBarSubSection>
+                    <CustomAvatar
+                        size={160}
+                        photoUrl={photoUrl}
+                        name={
+                            data?.corpUser?.editableProperties?.displayName ||
```

Comment:
Can we compute this once above then reuse it whenever we need a display name below?

Reply:
sure